source/common/stats/timespan_impl.cc

```diff
@@ -0,0 +1,47 @@
+#include "common/stats/timespan_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+HistogramCompletableTimespanImpl::HistogramCompletableTimespanImpl(Histogram& histogram,
+                                                                   TimeSource& time_source)
+    : time_source_(time_source), histogram_(histogram), start_(time_source.monotonicTime()) {
+  if (!histogram.active()) {
+    return;
+  }
+
+  switch (histogram.unit()) {
```

Comment:
@jmarantz I was talking about this switch, histogram can define another enum say `Quantity` and we can add a util method to derive `quantity` from `unit` so code would only need to do
`ASSERT(Util::quantity(histogram.unit()) == Histogram::Quantity::Time)`

or even add it to the histogram interface itself.

worth it or leave as-is?

Reply:
You could also just do
```
  ASSERT(histogram.unit() == Histogram::Unit::Microseconds || histogram.unit() == Histogram::Unit::Milliseconds);
```
that is (a) less lines of code than what you have and (b) will print out exactly what the problem is in the assertion method if it fails. If there was some other reason to create Util::quantity() you could use that, but I wouldn't create it for this.

Reply:
Thanks for your take on this, I made it a switch since a modern compiler checks if all cases were tested so if a new time unit is added, the compilation would fail and it would be obvious what places need to be updated, if I just check for micros and millis here and one day someone adds nanos, this will introduce a silent bug. No strong preference either way, just let me know if you still prefer ` ASSERT(histogram.unit() == Histogram::Unit::Microseconds || histogram.unit() == Histogram::Unit::Milliseconds);` compared to a full switch and I'll update it..

Reply:
Great point. I agree.